common/persistence/client/factory.go

```diff
@@ -73,18 +73,25 @@ type (
 		NewVisibilityStore() (p.VisibilityStore, error)
 		NewQueue(queueType p.QueueType) (p.Queue, error)
 	}
+	// AbstractDataStoreFactory creates a DataStoreFactory, can be used to implement custom datastore support outside
+	// of the cadence core.
+	AbstractDataStoreFactory interface {
```

Comment:
Maybe rename as CustomDatastoreFactory? (to be consistent with CustomDatastoreConfig)

Reply:
What this thing does falls exactly under the definition of an abstract factory pattern, hence IMO this is the least confusing name for it.